src/talker_npc.cpp

```diff
@@ -0,0 +1,818 @@
+    }
+
+    add_topics.push_back( me_npc->chatbin.first_topic );
+    if( radio_contact ) {
+        add_topics.push_back( "TALK_RADIO" );
+    } else if( me_npc->is_leader() ) {
+        add_topics.push_back( "TALK_LEADER" );
+    } else if( me_npc->is_player_ally() && ( me_npc->is_walking_with() || me_npc->has_activity() ) ) {
+        add_topics.push_back( "TALK_FRIEND" );
+    } else if( me_npc->get_attitude() == NPCATT_RECOVER_GOODS ) {
+        add_topics.push_back( "TALK_STOLE_ITEM" );
+    }
+    int most_difficult_mission = 0;
+    for( auto &mission : me_npc->chatbin.missions ) {
+        const auto &type = mission->get_type();
+        if( type.urgent && type.difficulty > most_difficult_mission ) {
+            add_topics.push_back( "TALK_MISSION_DESCRIBE_URGENT" );
+            me_npc->chatbin.mission_selected = mission;
+            most_difficult_mission = type.difficulty;
+        }
+    }
+    most_difficult_mission = 0;
+    bool chosen_urgent = false;
+    for( auto &mission : me_npc->chatbin.missions_assigned ) {
+        if( mission->get_assigned_player_id() != g->u.getID() ) {
+            // Not assigned to the player that is currently talking to the npc
+            continue;
+        }
+        const auto &type = mission->get_type();
+        if( ( type.urgent && !chosen_urgent ) || ( type.difficulty > most_difficult_mission &&
+                ( type.urgent || !chosen_urgent ) ) ) {
+            chosen_urgent = type.urgent;
+            add_topics.push_back( "TALK_MISSION_INQUIRE" );
+            me_npc->chatbin.mission_selected = mission;
+            most_difficult_mission = type.difficulty;
+        }
+    }
+
+    // Needs
+    if( me_npc->has_effect( effect_npc_suspend ) ) {
+        add_topics.push_back( "TALK_REBOOT" );
+    }
+    if( me_npc->has_effect( effect_sleep ) || me_npc->has_effect( effect_lying_down ) ) {
+        if( me_npc->has_effect( effect_narcosis ) ) {
+            add_topics.push_back( "TALK_SEDATED" );
+        } else {
+            add_topics.push_back( "TALK_WAKE_UP" );
+        }
+    }
+
+    if( add_topics.back() == "TALK_NONE" ) {
+        add_topics.back() = me_npc->pick_talk_topic( g->u );
+    }
+    me_npc->moves -= 100;
+
+    if( g->u.is_deaf() ) {
+        if( add_topics.back() == "TALK_MUG" ||
+            add_topics.back() == "TALK_STRANGER_AGGRESSIVE" ) {
+            me_npc->make_angry();
+            add_topics.push_back( "TALK_DEAF_ANGRY" );
+        } else {
+            add_topics.push_back( "TALK_DEAF" );
+        }
+    }
+
+    if( me_npc->has_trait( trait_PROF_FOODP ) &&
+        !( me_npc->is_wearing( itype_id( "foodperson_mask_on" ) ) ||
+           me_npc->is_wearing( itype_id( "foodperson_mask" ) ) ) ) {
+        add_topics.push_back( "TALK_NPC_NOFACE" );
+    }
+    me_npc->decide_needs();
+
+    return add_topics;
+}
+
+std::string talker_npc::get_job_description() const
+{
+    return me_npc->describe_mission();
```

Comment:
Why the terminology change from mission to job?

Reply:
Because the underlying NPC code is stupid.
An NPCMISSION is their current thing, like guarding or performing an activity, which is different than a player mission.

For the generic interface, job_description is probably more appropriate, even if it is implemented by the `talker_npc` code by calling `describe_mission()`.